src/lib/bootloader/mbr_update.rb

```diff
@@ -167,64 +177,73 @@ def bios_id_for(device)
 
     # List of partition for disk that can be used for setting boot flag
     def activatable_partitions(disk)
-      partitions = target_map.fetch(disk, {}).fetch("partitions", [])
+      return [] unless disk
+
       # do not select swap and do not select BIOS grub partition
       # as it clear its special flags (bnc#894040)
-      partitions.select do |p|
-        p["used_fs"] != :swap && p["fsid"] != Yast::Partitions.fsid_bios_grub
-      end
+      disk.partitions.reject { |p| p.id.is?(:swap, :bios_boot) }
     end
 
-    def extended_partition_num(disk)
-      part = activatable_partitions(disk).find { |p| p["type"] == :extended }
+    def extended_partition(partition)
+      part = partition.partitionable.partitions.find { |p| p.type.is?(:extended) }
       return nil unless part
 
-      num = part["nr"]
-      log.info "Using extended partition #{num} instead"
-      num
+      log.info "Using extended partition instead: #{part.inspect}"
+      part
     end
 
-    # Given a device name to which we install the bootloader (loader_device),
-    # gets back disk and partition number to activate. If empty Hash is returned
-    # then no suitable partition to activate found.
-    # @param [String] loader_device string the device to install bootloader to
-    # @return a Hash `{ "mbr" => String, "num" => Integer }`
-    #  containing disk (eg. "/dev/hda") and partition number (eg. 4)
+    # Given a device name (the bootloader location), returns the partition
+    # to activate.
+    #
+    # Raises an exception if no suitable partition to activate was found.
+    #
+    # @param loader_device [String] the device to install the bootloader to
+    #
+    # @return [Y2Storage::Partition]
+    #
     def partition_to_activate(loader_device)
+      # storage-ng
+      # FIXME
+      # going through 'real' device(s) here is almost certainly wrong; atm the
```

Comment:
well, tricky part is that we still does not use udev device names here. So if we passing around storage object and not device name, it can just work.